x-pack/platform/plugins/private/snapshot_restore/public/application/components/disable_tooltip.tsx

```diff
@@ -42,12 +42,16 @@ export const DisableToolTip: React.FunctionComponent<Props> = ({
   isManaged,
   tooltipMessage,
   component,
+  ...rest
 }) => {
+  // Ensures that any additional props (e.g. id, aria-describedby) are forwarded to the wrapped component
+  const componentWithProps = cloneElement(component, { ...rest });
```

Comment:
In my experience this approach can lead to prop element and component itself having competing props, which can lead to unintentional overrides of props

Do you see anything wrong with original behavior with strict narrow prop interface and forcing the consumer to add props directly on the component element?